test/source/allocation.cpp

```diff
@@ -19,13 +19,79 @@ auto get_one(smeagle::Corpus const& corpus, char const* name) {
   return funcs;
 }
 
-TEST_CASE("Register Allocation") {
+TEST_CASE("Register Allocation and Types") {
   auto corpus = smeagle::Smeagle("liballocation.so").parse();
 
   SUBCASE("one int") {
-    auto funcs = get_one(corpus, "f1");
+    auto funcs = get_one(corpus, "test_type_int");
     REQUIRE(funcs.size() == 1);
     auto const& parameters = funcs[0].parameters;
+    CHECK(parameters[0].type == "int");
+    CHECK(parameters[0].location == "%rdi");
+  }
+
+  SUBCASE("one string") {
```

Comment:
This is an aggregate type (a user-defined class) which we don't handle yet.

Reply:
There's a lot of types we don't handle - right now it just returns two framebases. I'm okay to leave them as templates and modify when we have the implementation! Or at least for the ones I've already written (and then would have to write again).